7TV emote support + Chat parser support for multiple emote providers

bryanrafaelbueno opened the pull request:
This PR adds support for **7TV emotes** and integrates them into the Twitch chat.

Emotes are fetched directly from the 7TV API and rendered alongside native Twitch emotes.

---

## **What’s inside**

* 7TV emote fetching via new backend endpoint
* Automatic loading of 7TV emotes per Twitch channel
* Chat parser support for multiple emote providers
* Proper emote source handling (`twitch` / `7tv`)
* Twitch emotes take priority on name conflicts

---

## **How to test**

* Run the app
* Open a Twitch channel with 7TV emotes enabled
* Send messages containing 7TV emotes
* Verify Twitch and 7TV emotes render correctly together

---

## **TODO**

* Improve 7TV emote caching
* Add support for BTTV / FFZ emotes

Commit:
feat: add support for 7TV emotes and integrate fetching from 7TV API

Commit:
docs: Update CHANGELOG.md to 0.0.3 version

Dan-Mizu commented:
Great work so far.

I see that zero-width emotes aren't working yet. They're the emotes that overlay the ones before them.

PeepoChat:
<img width="313" height="67" alt="firefox_os9a6q0TVB" src="https://github.com/user-attachments/assets/6b688595-49a9-4e12-aa22-6615bafa57e4" />

Chatterino:
<img width="327" height="71" alt="image" src="https://github.com/user-attachments/assets/5065d510-3618-4e0e-a381-a8af10f3069c" />

_In case you didn't know, you can press `Preview` under `Review` from the comment from Vercel to check out the deployment of the app online._

Commit:
feat: enhance emote handling with support for 0-width emotes in MessageContainer

bryanrafaelbueno commented:
Implemented the requested changes.
Please let me know if this matches what you needed.

Dan-Mizu commented:
Chatterino:
<img width="264" height="77" alt="image" src="https://github.com/user-attachments/assets/2e434076-6c69-402c-b557-316d8b241275" />

PeepoChat:
<img width="220" height="71" alt="image" src="https://github.com/user-attachments/assets/fa648681-3e86-4b3d-8e6f-4bc41dbbacec" />

Looks like normal zero width emotes work great, but ones that are wider have a sizing issue.

Commit:
feat: improve emote rendering logic to handle 0-width wide emotes

Commit:
feat: implement fetching and merging of global and user 7TV emotes

Commit:
docs: update CHANGELOG.md for version 0.0.4

Dan-Mizu committed:
Update package.json

Dan-Mizu committed:
fix: zero width emote layering on top of tooltips

Dan-Mizu committed:
fix: several stacked zero-width 7tv emotes

Dan-Mizu commented:
Still some issues that need to be addressed.

Need to get tooltips working with zero width stacked emotes like they do for single emotes.

Example of how its done in chatterino:
<img width="181" height="262" alt="image" src="https://github.com/user-attachments/assets/8ab9a38b-6ff1-464c-88db-054737949ef6" />

Something else not entirely related to emotes is when a message is several components and not just a single text component (any message with an emote in it) it doesn't wrap properly like a single text component does when the window width is skinny.

Some loading issues as well where hovering an emote doesn't load the image until you re-hover it or some console spamming when a ton of emotes are coming in at once. (easily test with erobb221's chat as they're constantly typing a ton)

Commit:
feat: Making multiple 0-width emotes work and tooltip support to 0-width emotes

Commit:
Merge branch 'preview' of https://github.com/GeleiaComPepino/PeepoChat into preview

Commit:
fix: improve handling of zero-width emotes to make it break lines

Commit:
docs: Update CHANGELOD.md

Dan-Mizu commented:
I'll have to do some cleanup as some issues that I noticed while developing has returned, like spaces between words if there are emotes in a message disappearing. I can't currently work on it, just reviewing it. Seems Vercel's build logs are also complaining about there being two Emote.vue components (the one in the badges/ directory shouldn't exist anymore- maybe a bad merge)

Dan-Mizu committed:
fix: emote rendering, tooltip, and popover issues

- Remove duplicate Emote.vue in components/badges that collided with components/emotes/Emote.vue and caused build warnings
- Render emotes with plain img tags instead of NuxtImg, since the IPX proxy rejects external CDN hosts (403) and caused failed emote loads and console spam during heavy chat
- Restore word spacing and line wrapping in messages containing emotes by emitting explicit space text nodes that Vue's whitespace condensing cannot drop
- Size zero-width emote stacks by their base emote and absolutely center overlays, so wide zero-width emotes spill over symmetrically without pushing surrounding text
- Add a Chatterino-style tooltip to stacked emotes that lists the base emote and every zero-width overlay
- Standardize tooltip emote previews to a fixed 64px height for parity between single and stacked tooltips, and reserve the image space so tooltips open at their final size instead of drifting off screen after the preview loads
- Apply a shared frosted-glass panel (80% opacity with backdrop blur) to all tooltips and dropdown menus via app.config, replacing per-component background overrides
- Enforce a single visible popover app-wide via a new popovers composable: opening a sidebar menu closes any other menu, a newly shown tooltip closes open menus, and menus close when the mouse leaves them
- Consolidate changelog entries under Unreleased and align the package version back to the last release (0.0.2)
